classes/robot/crawler.php

```diff
@@ -258,23 +258,25 @@ public function reset_for_recrawl($nodeid) {
             // Mark all nodes that link to this as needing a recrawl.
             if ($DB->get_dbfamily() == 'mysql') {
                 $DB->execute("UPDATE {tool_crawler_url} u
-                         INNER JOIN {tool_crawler_edge} e ON e.a = u.id
-                         SET needscrawl = ?,
-                                 lastcrawled = null
-                         WHERE e.b = ?", [$time, $nodeid]);
+                              INNER JOIN {tool_crawler_edge} e ON e.a = u.id
+                                     SET needscrawl = ?,
+                                         lastcrawled = null,
+                                         priority = ?
+                                   WHERE e.b = ?", [$time, TOOL_CRAWLER_PRIORITY_HIGH, $nodeid]);
             } else {
                 $DB->execute("UPDATE {tool_crawler_url} u
-                             SET needscrawl = ?,
-                                 lastcrawled = null
-                            FROM {tool_crawler_edge} e
-                           WHERE e.a = u.id
-                             AND e.b = ?", [$time, $nodeid]);
+                                     SET needscrawl = ?,
+                                         lastcrawled = null,
+                                         priority = ?
+                                    FROM {tool_crawler_edge} e
+                                   WHERE e.a = u.id
+                                     AND e.b = ?", [$time, TOOL_CRAWLER_PRIORITY_HIGH, $nodeid]);
             }
 
             // Delete all edges that point to this node.
             $DB->execute("DELETE
-                          FROM {tool_crawler_edge}
-                          WHERE b = ?", array($nodeid));
+                                FROM {tool_crawler_edge}
```

Comment:
also odd indenting, but also why is this an execute() and not just a delete_records() ? (not your patch but if you are touching it at all then why not fix it)

Reply:
Changed